packages/react-native-navigation/src/rum/instrumentation/DdRumReactNativeNavigationTracking.tsx

```diff
@@ -8,19 +8,26 @@ import React from 'react'
 import { ComponentDidAppearEvent, Navigation } from 'react-native-navigation';
 import { DdRum } from '@datadog/mobile-react-native';
 
+
+
+
+export type ViewNamePredicate = (trackedView: any, trackedName: string) => string | null
```

Comment:
why do we allow `null` return? what can be use-case for that?

Reply:
In case you only want to rename some views and keep the default value for others. You will return `null` for the ones you want to keep the default name.

Reply:
but there is `trackedName` argument which can be returned in this case, this is default name.

Reply:
oh indeed, there is no need for that yes

Reply:
Because this is JS, so we need to be prepared for people missusing this and returning null / undefined.

Reply:
If we want to protect ourselves from such case, maybe it is better to remove `null` from return type, so that TS type checker can highlight user code returning `undefined/null` where it can prove it (I think even in case of JS some IDEs can infer that), but still keep null-check in our code for the returned value? Otherwise we are saying that returning `null` is legit, and customer may think that by returning `null` view will be discarded. wdyt?

Reply:
My point is that to protect from such cases there is no need to add `null` to the return type definition. We can remove it and let IDE/compiler type checking do the job (even if JS is used it will work with decent IDE). And if user still breaks it ignoring IDE/compiler warnings and return `null`|`undefined`, then we can still keep `predicate(...) ?? defaultValue` line in the code, TS won't complain about it even if `predicate` doesn't have `null` in the definition.

Reply:
i agree with @0xnm, this is misleading: i thought returning `undefined` would drop the view

Reply:
> it will work with decent IDE
That's what I'm afraid of. 
But if you insist on it I'll mark it as non null. It just feels dirty to mark it as non-null when people could still return null without any warning. Marking it as null at least forces us to remember that the value could actually be undefined